### PR TITLE
Edit & Archive notes

### PR DESCRIPTION
**Pull request by jeetbhalani15**
## Added Edit & Archive Notes Functionality.
- Added Edit notes using API calls.
- Added Archive notes using API calls
  => Add & Delete Archive Notes.
- update badges in the sidebar menu.

Using hard-coded colors will update it after dark mode.

### CODE REVIEW

File: Gravity-Notes/src/Components/TextArea.js/Textarea.js [hunk 1]
```diff
@@ -68,16 +60,25 @@ export const Textarea = () => {
...
+         }
+        
 
       </form>
 
     {/* Note card component */}
 
-          {console.log(noteState.notes)}
-          
+          {/* {console.log(noteState.notes)}
```
Comment (Pratham82) [inline Gravity-Notes/src/Components/TextArea.js/Textarea.js:79]: Removed unused logs

**Review by Pratham82 — APPROVED** [review]
Working is good. no issues.

Some UI Suggestions:

- Add some padding between the actions
image

- Instead of using the vanilla js alert method use 3rd party alerts (if allowed) of custom alerts in comp library.
- Fix the eye icon on sign up page
image

Apart from these changes, the UI looks very nice and identical to google keep, good work.

File: Gravity-Notes/src/Components/TextArea.js/Textarea.js (hunk 1 above)
Comment (jeetbhalani15) [reply Gravity-Notes/src/Components/TextArea.js/Textarea.js:79]: sure  & I will make these changes in UI soon, was going for functionality first and soon gonna refactor my UI

thanks, bro🙌

File: Gravity-Notes/src/Components/Archive-card/ArchiveCard.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import {HiOutlineUserAdd} from "react-icons/hi"
+import {BiImage} from "react-icons/bi"
+import {BiArchiveIn} from "react-icons/bi"
+import {MdMoreVert} from "react-icons/md"
+import {BiEditAlt} from "react-icons/bi"
+import { useAuth } from "../../Contexts/Auth-context"
+import { useNotes } from "../../Contexts/NotesAction-context"
+import axios from "axios"
+import { deleteNote } from "../../Utils/DeleteNote"
+import { restoreArchiveNote } from "../../Utils/RestoreArchiveNotes"
```
Comment (hiteshverma27) [inline Gravity-Notes/src/Components/Archive-card/ArchiveCard.js:12]: use barrel exports

File: Gravity-Notes/src/Components/Archive-card/ArchiveCard.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { deleteFromArchive } from "../../Utils/DeleteFromArchive"
+
+export const ArchiveCard = ({ archive })=>{
+    const { noteDispatch } = useNotes();
+    const {authState} = useAuth();
+    return(
+        <div className='note'>
+        <div className="title-box">
+        <h1 className="title-txt">{archive.title}</h1>
+        <small>{archive.data}</small>
```
Comment (hiteshverma27) [inline Gravity-Notes/src/Components/Archive-card/ArchiveCard.js:22]: Nice use of semantic tags

File: Gravity-Notes/src/Components/Sidebar/Sidebar.css [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    justify-content: center;
+    align-items: center;
+}
+.tag{
+    font-size: small;
+    position: absolute;
+    top: 80%;
+    -webkit-transition: all .6s ease;
+    transition: all .6s ease;
+    /* visibility: hidden; */
```
Comment (hiteshverma27) [inline Gravity-Notes/src/Components/Sidebar/Sidebar.css]: again remove commented code before pushing

**Review by hiteshverma27 — APPROVED** [review]
Left few suggestions and rest looks good, some good things I noticed
- Good folder structure 
- Try barrel exports to make code more readable and manageable
- Clean and simple implementation of reducer function

**Pull request merged**